db/revision_cache_lru.go

```diff
@@ -307,15 +299,16 @@ func (rc *LRURevisionCache) getValue(docID, revID string, collectionID uint32, c
 		rc.cache[key] = rc.lruList.PushFront(value)
 		rc.cacheNumItems.Add(1)
 
-		// evict if over number capacity
-		var numItemsRemoved int
-		for len(rc.cache) > int(rc.capacity) {
-			rc.purgeOldest_()
-			numItemsRemoved++
+		numItemsRemoved, numBytesEvicted := rc._numberCapacityEviction()
+		if numBytesEvicted > 0 {
+			rc.updateRevCacheMemoryUsage(-numBytesEvicted)
 		}
+		rc.lock.Unlock() // release lock as eviction is finished
```

Comment:
As above, we can move unlock since `updateRevCacheMemoryUsage` doesn't require a lock